public_html/admin/controller/pages/sale/coupon.php

```diff
@@ -638,14 +650,29 @@ private function _getForm()
             );
         }
 
+        $this->loadModel('catalog/category');
+        $this->data['categories'] = [];
+        $allStores = $this->model_setting_store->getStores();
+        $results = $this->model_catalog_category->getCategories(0);
```

Comment:
Can we start using constants instead of meaningless numbers?
ROOT_CATEGORY_ID = 0

Reply:
done